File: e-shoping-frontend/frontend/src/app/Model/CustomerDetails/customerDetails.ts
export interface User{
firstName:String
lastName:String
phoneNo:String
email:String
username:String
}
export interface Customer{
id:number;
fullName:String;
phoneNo:String;
pinCode:number;
state:String;
city:String;
houseNo:String;
areaName:String;
typeOfAddress:String;
}
export class customerDetails{
customer!:Customer;
user!:User
constructor(customer:Customer,user:User){
this.customer=customer;
this.user=user
}
}

File: e-shoping-backend/backend/src/main/java/com/example/backend/entities/CustomerDetails.java
package com.example.backend.entities;

import com.fasterxml.jackson.annotation.JsonBackReference;
import jakarta.persistence.*;
import lombok.*;

@Entity
@Table(name = "customer_details")
@AllArgsConstructor
@NoArgsConstructor
@Getter
@Setter
public class CustomerDetails {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    private String fullName;
    private String phoneNo;
    private int pinCode;
    private String state;
    private String city;
    private String houseNo;
    private String areaName;
    private String typeOfAddress;
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "user_id")
    @JsonBackReference // Breaks the circular reference
    private User user;
}

